### DISCUSSION TITLE
How to prevent Nude pictures?

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#2942] [Q&A] [answered]
I set the prompt as "beautiful woman in a bedroom  beautiful woman", faceswap as option. 
I also set the negative prompt as "Nude, NSFW"

But it still shows the nude piccure. 

If there is anyone who can help me? Don't hesitate to let me know. Thank you.

**Comment by ILikeToasters** [comment, marked as answer]
You could add more weight to the words in the negative for one thing. "(nude:1.5)" for example. Also mention clothing. Part of the problem is the association effect and using "bedroom" in the prompt will have amore likelihood than if you said kitchen since people are more likely to be nude in the bedroom especially since you do not mention any clothing. Of course you could use mashb1t's fork which I know has a filter option to try an block those images. That might be included in a future update but have not looked.

**Reply by mashb1t** [reply]
https://github.com/lllyasviel/Fooocus/pull/958 requires a python package update (dependency to diffusers), but this might be extracted. Planned for milestone https://github.com/lllyasviel/Fooocus/milestone/4.

**Reply by oldhand7** [reply]
So Do I have to set the negative prompt as "(nude:1.5), naked, distorted"?


How to adapt the censoring function to existing codebase?

**Reply by ILikeToasters** [reply]
The added weight can be used but really your best bet is to avoid having to use them by using a good regular prompt. Just describe some clothing in your prompt. If you insist on not giving her clothing she will probably be nude.

**Reply by ILikeToasters** [reply]
And I would also recommend checking out some YouTube videos or reading some tutorials about Stable Diffusion and Fooocus to understand how things work. If you did you would probably not have so many questions. I am not trying to be rude but most of the questions you ask are usually answered if you look some of that stuff up. The answers are already out there.
